### DISCUSSION TITLE
API INVOICE - CreateIssuedDocumentRequest - CedentePrestatore - Regime Fiscale

### DISCUSSION THREAD

**Discussion by Mustino2022** [fattureincloud/api#335] [Q&A] [answered]
Buongiorno,
nella creazione del documento tramite le API non riesco a trovare la proprietà dove impostare il Regime Fiscale.

Anche nella libreria Fatture in Cloud C# SDK non riesco a trovare dove si imposta il Regime Fiscale (RF17 o RF11)

Nel XML è presente in CedentePrestatore > DatiAnagrafici > RegimeFiscale

Restando in attesa di riscontro, ringrazio e saluto cordialmente

**Comment by emajo** [comment, marked as answer]
Ciao @Mustino2022, per impostare il regime fiscare è necesario utilizzare il campo `ei_raw` come riportato di seguito:
```
        "ei_raw": {
            "FatturaElettronicaHeader": {
                "CedentePrestatore": {
                    "DatiAnagrafici": {
                        "RegimeFiscale": "RF17"
                    }
                }
            }
        }
```
abbiamo anche[ una guida ](https://developers.fattureincloud.it/docs/guides/e-invoice-xml-customisation)che spiega l'utilizzo di questo campo e contiene anche esempi per le varie SDKs.

Grazie,
Emanuele

**Comment by Mustino2022** [comment]
Buongiorno, grazie per la risposta, quindi il codice c# sarebbe cosi, è corretto?

IssuedDocument invoice = new IssuedDocument(
    type: IssuedDocumentType.Invoice,
    eiRaw: new {
        FatturaElettronicaHeader= new {
            CedentePrestatore= new {
                DatiAnagrafici= new [] {
                    new {
                        RegimeFiscale= "RF17"
                    }
                }
            }
        }
    }
);

**Comment by Mustino2022** [comment]
Ho inserito il codice ma purtroppo ricevo il seguente errore

Validation Result: Nei dati del Cedente Prestatore (fornitore), all'interno dell'elemento DatiAnagrafici dovrebbe esserci l'elemento IdFiscaleIVA
Nei dati del Cedente Prestatore (fornitore), all'interno dell'elemento CedentePrestatore dovrebbe esserci l'elemento Sede

In realtà quei dati sono già valorizzati, ma con l'inserimento nel nuovo codice ora si presenta questo problema.

Io non vorrei inserire una nuova eiRaw completa ma indicare solamente il campo RegimeFiscale, è possibile?

**Comment by emajo** [comment]
il codice dovrebbe essere così:

```
IssuedDocument invoice = new IssuedDocument(
    type: IssuedDocumentType.Invoice,
    eiRaw: new {
        FatturaElettronicaHeader= new {
            CedentePrestatore= new {
                DatiAnagrafici= new {
                    RegimeFiscale= "RF17"
                }
            }
        }
    }
);
```

risolve il problema?

**Comment by Mustino2022** [comment]
Salve @emajo funziona, grazie mille, buon lavoro
